src/profile-logic/profile-data.js

```diff
@@ -1940,12 +1940,10 @@ export function getSampleIndexClosestToTime(
   // and its predecessor.
   const previousIndex = index - 1;
 
-  let weight = interval;
-  let previousWeight = interval;
-  if (samples.weight) {
-    weight = Math.abs(samples.weight[index]);
-    previousWeight = Math.abs(samples.weight[previousIndex]);
-  }
+  const weight = samples.weight ? Math.abs(samples.weight[index]) : interval;
+  const previousWeight = samples.weight
+    ? Math.abs(samples.weight[previousIndex])
+    : interval;
```

Comment:
It's a bit weird that we didn't have this problem earlier, but well, this new code works for me too.

Reply:
True. Strangely, flow was only flagging the second line in the if condition for accessing an index for a null object, and not the first line, or any other references to SamplesTable weight in the project

Reply:
Flow assumes that function calls can change properties, which invalidates the if check. Math.abs is a function call so here's the reason. Another way to fix it is to assign samples.weight to a const and use that const, because const can't be reassigned so flow knows it's not changed even after the function calls.

Reply:
I noticed that we got this flow error after we changed the implementation of bisection. The initial implementation that came from the library did not cause this.